graphql_compiler/query_pagination/pagination_planning.py

```diff
@@ -1,53 +1,174 @@
 # Copyright 2019-present Kensho Technologies, LLC.
-from collections import namedtuple
+from dataclasses import dataclass, field
+from typing import List, NamedTuple, Tuple
+
+from graphql import DocumentNode
 
 from ..ast_manipulation import get_only_query_definition, get_only_selection_from_ast
-from ..cost_estimation.helpers import is_int_field_type, is_uuid4_type
+from ..cost_estimation.helpers import is_uuid4_type
+from ..cost_estimation.int_value_conversion import field_supports_range_reasoning
 from ..exceptions import GraphQLError
+from ..schema.schema_info import QueryPlanningSchemaInfo
+
+
+@dataclass
+class PaginationAdvisory(object):
+    message: str
+
+
+@dataclass
+class PaginationFieldNotSpecified(PaginationAdvisory):
+    vertex_name: str
+    message: str = field(init=False)
```

Comment:
I don't think this is required. `message` exists already due to the parent class, and doesn't need a redefinition. I think this is shadowing the parent class' field.